swarmrl/intrinsic_reward/random_network_distillation.py

```diff
@@ -0,0 +1,138 @@
+"""
+Reward functions based on Random Network Distillation.
+
+Notes
+-----
+https://arxiv.org/abs/1810.12894
+"""
+
+import jax.numpy as np
+from znnl.models.flax_model import FlaxModel
+from znnl.models.jax_model import JaxModel
+
+from swarmrl.intrinsic_reward.intrinsic_reward import IntrinsicReward
+from swarmrl.intrinsic_reward.rnd_configs import RNDArchitecture, RNDConfig
+from swarmrl.utils.colloid_utils import TrajectoryInformation
+
+
+class RNDReward(IntrinsicReward):
+    """
+    Class to implement an intrinsic loss based on Random Network Distillation.
+
+    This implementation is based on the neural networks framework implemented in
+    the ZnNL library.
+
+    Attributes
+    ----------
+
+    """
+
+    def __init__(self, rnd_config: RNDConfig):
+        """
+        Constructor for the RND Loss class.
+
+        Parameters
+        ----------
+        See RNDConfig for more information.
+        """
+        # Automatically unpack the configuration.
+        self.__dict__.update(rnd_config.__dict__)
+
+        self.iterations = 0
+        self.metric_results = None
+
+        # Initialize the predictor into the training strategy.
+        self.target_network: JaxModel = FlaxModel(
+            flax_module=RNDArchitecture(),
+            optimizer=rnd_config.optimizer,
+            input_shape=(1, *rnd_config.input_shape),
+        )
+        self.predictor_network: JaxModel = FlaxModel(
+            flax_module=RNDArchitecture(),
+            optimizer=rnd_config.optimizer,
+            input_shape=(1, *rnd_config.input_shape),
+        )
+        self.training_strategy.set_model(self.predictor_network)
+
+    @staticmethod
+    def _reshape_data(x: np.ndarray) -> np.ndarray:
+        """
+        Reshape the data to treat the n_steps and num_colloids equally.
```

Comment:
Perhaps this comment can read "Flatten the first two axes of the array". I don't find the current wording clear. If I didn't read the code, I would not understand what "treat them equally" means.

Reply:
changed in [981cd2e](https://github.com/SwarmRL/SwarmRL/pull/75/commits/981cd2e708a7bcc20657e7e1a0eb5453f6286d88)